RocketChat/Rocket.Chat
# Regression: Bad words Filtering not working on new Message Parser

## Proposed changes (including videos or screenshots)

Was
![forGH](https://user-images.githubusercontent.com/118266546/204087809-fca6152b-63db-4873-88b8-d86aeb324531.png)

Now
![forGH2](https://user-images.githubusercontent.com/118266546/204087822-e3273127-a4f0-4a45-a73e-910eb48c9b6f.png)






## Issue(s)

closes https://github.com/RocketChat/Rocket.Chat/issues/27298
## Steps to test or reproduce


1.) Turn off legacy message template (if on). `Account -> Preferences -> global -> Use legacy message template`
2.) Turn on filtering of bad words `Administration -> settings -> message -> Allow Message bad words filtering`
3.) Add bad words to the bad words list (comma separated) `Administration -> settings -> message -> Add Bad Words to the Blacklist`
4.) Write a message with bad words. They should be censored.

## Further comments

The bug happened because of precedence of message markdown parsing was higher than message censoring.
